### ISSUE TITLE
RegEx issues

### ISSUE THREAD

**Issue by JohnMurga** [svaarala/duktape#555]
In code taken from Mocha and Chai :
```javascript
var re1 = /^function *\(.*\)\s*{|\(.*\) *=> *{?/
var re2 = /#{this}/g
var re3 = new RegExp("/#{this}/g")
var re4 = new RegExp("/^function *\(.*\)\s*{|\(.*\) *=> *{?/")
```
Seems to work on V8 and Gecko

On Duktape I get :
```
SyntaxError: invalid regexp quantifier (unknown char)
```

**Comment by fatcerberus** [comment]
This should have been fixed by #513.  Try it with the latest master, those regexps should work there.

**Comment by svaarala** [comment]
@JohnMurga The issue with those RegExps is that they use curly braces as literals without escaping, which is non-standard syntax, the fix would be to escape, e.g.:

    /{#this}/g  -->  /\{#this\}/g

But like @fatcerberus said, Duktape master now parses these non-standard regexps because they appear in quite many code bases.

**Comment by svaarala** [comment]
Closing this as a closed duplicate to #142.
